xds/src/main/java/io/grpc/xds/internal/certprovider/FileWatcherCertificateProvider.java

```diff
@@ -109,18 +80,19 @@ public void start() {
   }
 
   @Override
-  public void close() {
-    if (scheduledHandle != null) {
-      scheduledHandle.cancel();
-      scheduledHandle = null;
+  public synchronized void close() {
+    shutdown = true;
+    if (scheduledFuture != null) {
+      scheduledFuture.cancel(true);
+      scheduledFuture = null;
     }
     getWatcher().close();
```

Comment:
`getWatcher().close()` doesn't need be synchronized. Otherwise LGTM.

Reply:
true but looking at `CertificateProvider.DistributorWatcher.close()` (which is what this will call), it's fairly trivial and is not going to be affected by calling from a synchronized block.